MarvelApiClient/src/main/java/com/karumi/MarvelApiClient/MarvelApiClient.java

```diff
@@ -28,97 +29,155 @@
  */
 public class MarvelApiClient {
 
-    private static MarvelApiClient singleton;
-    private final String publicKey;
-    private final String privateKey;
-    private final boolean debug;
-    private final Retrofit retrofit;
-
-    MarvelApiClient(String publicKey, String privateKey, Retrofit retrofit, boolean debug) {
-        this.publicKey = publicKey;
-        this.privateKey = privateKey;
-        this.retrofit = retrofit;
-        this.debug = debug;
-    }
-
-    public static MarvelApiClient with(String publicKey, String privateKey) {
+  private static MarvelApiClient singleton;
+  private final String publicKey;
+  private final String privateKey;
+  private final boolean debug;
+  private final ApiCall apiCall;
+
+  MarvelApiClient(String publicKey, String privateKey, ApiCall apiCall, boolean debug) {
+    this.publicKey = publicKey;
+    this.privateKey = privateKey;
+    this.apiCall = apiCall;
+    this.debug = debug;
+  }
+
+  public static MarvelApiClient with(String publicKey, String privateKey) {
+    if (singleton == null) {
+      synchronized (MarvelApiClient.class) {
```

Comment:
There are simpler ways to create safe singletons, we don't need this level of optimization and it makes the code somewhat unreadable for people who don't know this specific pattern.